lib/ex_crypto.ex

```diff
@@ -211,18 +211,63 @@ defmodule ExCrypto do
   """
   @spec encrypt(binary, binary, binary, binary) :: {:ok, {binary, {binary, binary, binary}}} | {:error, binary}
   def encrypt(key, authentication_data, initialization_vector, clear_text) do
-    case :crypto.block_encrypt(:aes_gcm, key, initialization_vector, {authentication_data, clear_text}) do
-      {cipher_text, cipher_tag} -> {:ok, {authentication_data, {initialization_vector, cipher_text, cipher_tag}}}
+    _encrypt(key, initialization_vector, {authentication_data, clear_text}, :aes_gcm)
+  end
+
+
+  @doc """
+  Encrypt a `binary` with AES in CBC mode.
+
+  Returns a tuple containing the `initialization_vector`, and `cipher_text`.
+
+  At a high level encryption using AES in CBC mode looks like this:
+
+      key + clear_text -> init_vec + cipher_text
+
+  ## Examples
+
+      iex> clear_text = "my-clear-text"
+      iex> {:ok, aes_256_key} = ExCrypto.generate_aes_key(:aes_256, :bytes)
+      iex> {:ok, {iv, payload}} = ExCrypto.encrypt(aes_256_key, clear_text)
+      iex> {iv, cipher_text} = payload
+      iex> assert(is_bitstring(cipher_text))
+      true
+
+  """
+  @spec encrypt(binary, binary, binary) :: {:ok, {binary, {binary, binary}}} | {:error, binary}
+  def encrypt(key, clear_text) do
```

Comment:
I closed the other PR, and basically fixed this one, though I think it's just a jumping off point for our further refactors discussed here and there.  What do you think of encrypt and decrypt taking 2 and 3 args required respectively, and optional map for explicit iv, algorithm specification, maybe a few others things?

Also, I came across this https://github.com/RNCryptor/RNCryptor-erlang and wonder if there's any chance that's a "better" base dependency for us than erlang's native crypto, since it's already geared for interop, but thought I'd ping here before I explored that further.  Haven't tried it yet, but curious for impressions or if you have any experience.  Could also try and make underlying lib pluggable, to support base erlang, this lib and libsodium or something? but that's probably a further iteration and not sure if it's a good idea anyway...